### PR TITLE
planner: add more log and metrics for instance plan cache eviction

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #54057

Problem Summary: planner: add more log and metrics for instance plan cache eviction

### What changed and how does it work?

1. add more log and metrics for instance plan cache eviction
2. prevent updating instance plan cache items during eviction
3. update the default eviction interval from 15s to 30s

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: pkg/domain/domain.go [hunk 1]
```diff
@@ -3182,7 +3182,7 @@ func (do *Domain) planCacheMetricsAndVars() {
 // planCacheEvictTrigger triggers the plan cache eviction periodically.
 func (do *Domain) planCacheEvictTrigger() {
 	defer util.Recover(metrics.LabelDomain, "planCacheEvictTrigger", nil, false)
-	ticker := time.NewTicker(time.Second * 15) // 15s by default
+	ticker := time.NewTicker(time.Second * 30) // 30s by default
```
Comment (AilinKid) [inline pkg/domain/domain.go:3185]: this value is changed deliberately？

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

File: pkg/domain/domain.go (hunk 1 above)
Comment (qw4990) [reply pkg/domain/domain.go:3185]: YES. 15s seems too frequent for this feature.

**Review by AilinKid — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**